docs/BEMAN_STANDARD.md

```diff
@@ -422,3 +452,21 @@ copyright notice following the SPDX license identifier.
 
 **[CPP.NAMESPACE]** RECOMMENDATION: Headers in `include/beman/<short_name>/` should export
 entities in the `beman::<short_name>` namespace.
+
+**[CPP.NO_FLAG_FORKING]** REQUIREMENT: C++ preprocessing must produce identical
+output regardless of compiler flags.
+
+Therefore, feature test macros such as `__cpp_explicit_this_parameter` should
+not be used directly. Instead use the following approach for feature-dependent
+code generation:
+
+1. Check for availability at CMake time using, for example,
+   `check_cxx_source_compiles`
+2. Create a CMake `option` (e.g. `BEMAN_&lt;short_name&gt;_USE_DEDUCING_THIS`)
+   with a default value based on detected support.
+3. Generate a `config.hpp` with a `#define` macro set to the selected option.
+4. Use this macro in place of the feature test macro.
+
+See
+[beman.iterator_interface](https://github.com/bemanproject/iterator_interface/blob/5e6714e10faa1799723669e04abec6e75adbdb89/CMakeLists.txt#L44)
```

Comment:
Do you think it's better to have this example inplace in this doc? We can have collapsable sections as for exemplar docs

<img width="1025" alt="image" src="https://github.com/user-attachments/assets/128e0399-4d74-4577-962f-8e1178450301" />